### PR TITLE
add 0d tensor support for searchsorted for gpu and cpu

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Describe
0d tensor support for searchsorted

xpu and mlu: https://github.com/PaddlePaddle/PaddleCustomDevice/pull/233

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 1]
```diff
@@ -656,6 +665,16 @@ def test_logical_not(self):
         res = self.exe.run(prog, fetch_list=[out])
         self.assertEqual(res[0].shape, ())
 
+    @prog_scope()
+    def test_searchsorted(self):
+        x = paddle.full([5], 1.0, 'float32')
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: 这个建议把这个的shape设置大些

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 2]
```diff
@@ -586,6 +586,15 @@ def test_logical_not(self):
...
         self.assertEqual(out.shape, [])
 
+    def test_searchsorted(self):
+        x = paddle.to_tensor([1, 3, 5, 7, 9])
+        y = paddle.rand([])
+
+        # only has forward kernel
+        out = paddle.searchsorted(x, y)
+
+        self.assertEqual(out.shape, [])
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:596]: 可以判断下输出值恒为1

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 1 above)
Comment (FeixLiu) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: ok

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 2 above)
Comment (FeixLiu) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:596]: ok

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Pull request merged**